standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -12872,15 +12879,16 @@ public int deleteRuntimeStats(int maxRetainSecs) throws MetaException {
       return 0;
     }
     boolean committed = false;
-    try {
+    try (Query q = pm.newQuery(MRuntimeStat.class)) {
       openTransaction();
       int maxCreateTime = (int) (System.currentTimeMillis() / 1000) - maxRetainSecs;
-      Query q = pm.newQuery(MRuntimeStat.class);
       q.setFilter("createTime <= maxCreateTime");
       q.declareParameters("int maxCreateTime");
       long deleted = q.deletePersistentAll(maxCreateTime);
       committed = commitTransaction();
       return (int) deleted;
+    } catch (Exception e) {
+      throw new MetaException(e.getMessage());
```

Comment:
is this catch needed?

Reply:
A checked exception should be handled by try-with-resources

Reply:
```try (Query q = pm.newQuery(MRuntimeStat.class))``` throws a checked exception, will use ```try (QueryWrapper wrapper = new QueryWrapper())``` to eliminate the exception handling.